ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/ErgoInterpreter.scala

```diff
@@ -82,12 +87,62 @@ class ErgoInterpreter(params: ErgoLikeParameters)(implicit IR: IRContext)
 
 object ErgoInterpreter {
 
+  /** Creates an interpreter with the given parameters, [[RuntimeIRContext]] for script processing and
+    * the `scriptProcessor` for handling compilation caching and handling pre-compiled scripts.
+    */
   def apply(params: ErgoLikeParameters): ErgoInterpreter =
-    new ErgoInterpreter(params)(new RuntimeIRContext)
+    new ErgoInterpreter(params)(new RuntimeIRContext) {
+      override val precompiledScriptProcessor = scriptProcessor
+    }
 
+  /** Create [[AvlTreeData]] with the given digest and all operations enabled. */
   def avlTreeFromDigest(digest: ADDigest): AvlTreeData = {
     val flags = AvlTreeFlags(insertAllowed = true, updateAllowed = true, removeAllowed = true)
     AvlTreeData(digest, flags, Constants.HashLength)
   }
 
+  /** A list of settings, one for each soft-fork. */
```

Comment:
The ScalaDoc comment is not descriptive. Please provide more details.

Reply:
This is actually not necessary, removed.